### PR TITLE
Feature/add apple watch

### PR DESCRIPTION
**Pull request by NathanDVT**
- [X] Added Apple Watch
- [X] Removed unused files
- [X] Added UI Tests

### CODE REVIEW

File: realwork/WatchApp Extension/ExtensionDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import WatchKit
+
+class ExtensionDelegate: NSObject, WKExtensionDelegate {
+
+    func applicationDidFinishLaunching() {
+        // Perform any final initialization of your application.
+    }
+
+    func applicationDidBecomeActive() {
+        // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
```
Comment (NathanDVT) [inline realwork/WatchApp Extension/ExtensionDelegate.swift:18]: Codacy Issue found: [Line should be 120 characters or less: currently 194 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           

Clones removed
==============
+ realwork/realwork/SearchBoard/SearchVC.swift  -2
+ realwork/realwork/ViewController.swift  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460&bid=17852899)

File: realwork/WatchApp Extension/InterfaceController.swift [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import Foundation
+import WatchConnectivity //**1
+
+class InterfaceController: WKInterfaceController {
+
+    let session = WCSession.default//**3
+    let colorThemes: [UIColor] = [.yellow, .blue,
+                                .magenta, .red,
+                                .purple, .green
+
```
Comment (NathanDVT) [inline realwork/WatchApp Extension/InterfaceController.swift:19]: Codacy Issue found: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/realwork/Dashboard/HomeBoard/MusicBarViewController.swift [hunk 3]
```diff
@@ -45,5 +72,28 @@ class MusicBarViewController: UIViewController {
...
+
+extension MusicBarViewController: WCSessionDelegate {
+
+  func sessionDidBecomeInactive(_ session: WCSession) {
+  }
+
+  func sessionDidDeactivate(_ session: WCSession) {
+  }
+
+  func session(_ session: WCSession, activationDidCompleteWith activationState: WCSessionActivationState, error: Error?) {
```
Comment (NathanDVT) [inline realwork/realwork/Dashboard/HomeBoard/MusicBarViewController.swift:87]: Codacy Issue found: [Line should be 120 characters or less: currently 122 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/WatchApp Extension/NotificationController.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+//
+//  Created by Nathan Ngobale on 2020/04/29.
+//  Copyright © 2020 Nathan Ngobale. All rights reserved.
+//
+
+import WatchKit
+import Foundation
+import UserNotifications
+
+
```
Comment (NathanDVT) [inline realwork/WatchApp Extension/NotificationController.swift:13]: Codacy Issue found: [Limit vertical whitespace to a single empty line. Currently 2.](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/WatchApp Extension/NotificationController.swift [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+class NotificationController: WKUserNotificationInterfaceController {
+
+    override init() {
+        // Initialize variables here.
+        super.init()
+
```
Comment (NathanDVT) [inline realwork/WatchApp Extension/NotificationController.swift:19]: Codacy Issue found: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/WatchApp Extension/ExtensionDelegate.swift [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    func applicationWillResignActive() {
+        // Sent when the application is about to move from active to inactive state.
+        // This can occur for certain types of temporary interruptions (such as an incoming phone call
+        // or SMS message) or when the user quits the application
+        // and it begins the transition to the background state.
+        // Use this method to pause ongoing tasks, disable timers, etc.
+    }
+
+    func handle(_ backgroundTasks: Set<WKRefreshBackgroundTask>) {
+        // Sent when the system needs to launch the application in the background to process tasks. Tasks arrive in a set, so loop through and process each one.
```
Comment (NathanDVT) [inline realwork/WatchApp Extension/ExtensionDelegate.swift:30]: Codacy Issue found: [Line should be 120 characters or less: currently 160 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/WatchApp Extension/ExtensionDelegate.swift [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        for task in backgroundTasks {
+            // Use a switch statement to check the task type
+            switch task {
+            case let backgroundTask as WKApplicationRefreshBackgroundTask:
+                // Be sure to complete the background task once you’re done.
+                backgroundTask.setTaskCompletedWithSnapshot(false)
+            case let snapshotTask as WKSnapshotRefreshBackgroundTask:
+                // Snapshot tasks have a unique completion call, make sure to set your expiration date
+                snapshotTask.setTaskCompleted(restoredDefaultState: true, estimatedSnapshotExpiration: Date.distantFuture, userInfo: nil)
```
Comment (NathanDVT) [inline realwork/WatchApp Extension/ExtensionDelegate.swift:39]: Codacy Issue found: [Line should be 120 characters or less: currently 137 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

**Review by XanderSchoeman — APPROVED** [review]

**Review by JancoDVTDev — APPROVED** [review]

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           

Clones added
============
- realwork/realworkUITests/ScreenTests/SearchScreenTests.swift  7
         

Clones removed
==============
+ realwork/realwork/ViewController.swift  -2
+ realwork/realwork/SearchBoard/SearchVC.swift  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460&bid=17852899)

File: realwork/WatchApp Extension/ExtensionDelegate.swift (hunk 7 above)
Comment (NathanDVT) [inline realwork/WatchApp Extension/ExtensionDelegate.swift:39]: Codacy Issue found: [Line should be 120 characters or less: currently 137 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/realwork/Dashboard/HomeBoard/MusicBarViewController.swift (hunk 3 above)
Comment (NathanDVT) [inline realwork/realwork/Dashboard/HomeBoard/MusicBarViewController.swift:87]: Codacy Issue found: [Line should be 120 characters or less: currently 122 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/WatchApp Extension/NotificationController.swift (hunk 5 above)
Comment (NathanDVT) [inline realwork/WatchApp Extension/NotificationController.swift:19]: Codacy Issue found: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/WatchApp Extension/NotificationController.swift (hunk 4 above)
Comment (NathanDVT) [inline realwork/WatchApp Extension/NotificationController.swift:13]: Codacy Issue found: [Limit vertical whitespace to a single empty line. Currently 2.](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/WatchApp Extension/ExtensionDelegate.swift (hunk 6 above)
Comment (NathanDVT) [inline realwork/WatchApp Extension/ExtensionDelegate.swift:30]: Codacy Issue found: [Line should be 120 characters or less: currently 160 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/WatchApp Extension/InterfaceController.swift (hunk 2 above)
Comment (NathanDVT) [inline realwork/WatchApp Extension/InterfaceController.swift:19]: Codacy Issue found: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

File: realwork/WatchApp Extension/ExtensionDelegate.swift (hunk 1 above)
Comment (NathanDVT) [inline realwork/WatchApp Extension/ExtensionDelegate.swift:18]: Codacy Issue found: [Line should be 120 characters or less: currently 194 characters](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460)

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- realwork/realworkUITests/ScreenTests/SearchScreenTests.swift  7
         

Clones removed
==============
+ realwork/realwork/ViewController.swift  -2
+ realwork/realwork/SearchBoard/SearchVC.swift  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460&bid=17852899)

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- realwork/realworkUITests/ScreenTests/SearchScreenTests.swift  7
         

Clones removed
==============
+ realwork/realwork/ViewController.swift  -2
+ realwork/realwork/SearchBoard/SearchVC.swift  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460&bid=17852899)

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- realwork/realworkUITests/ScreenTests/SearchScreenTests.swift  7
         

Clones removed
==============
+ realwork/realwork/SearchBoard/SearchVC.swift  -2
+ realwork/realwork/ViewController.swift  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460&bid=17852899)

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- realwork/realworkUITests/ScreenTests/SearchScreenTests.swift  7
         

Clones removed
==============
+ realwork/realwork/SearchBoard/SearchVC.swift  -2
+ realwork/realwork/ViewController.swift  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/Musico/pullRequest?prid=5444460&bid=17852899)

**Pull request merged**